flymake-languagetool.el

```diff
@@ -65,163 +87,168 @@
-  (let ((matches (cdr (assoc 'matches flymake-languagetool--output)))
-        check-list)
-    (dolist (match matches)
-      (let* ((pt-beg (cdr (assoc 'offset match)))
-             (len (cdr (assoc 'length match)))
-             (pt-end (+ pt-beg len))
-             (type 'warning)
-             (desc (cdr (assoc 'message match))))
-        (push (flymake-make-diagnostic source-buffer (1+ pt-beg) (1+ pt-end) type desc) check-list)))
-    (progn  ; Remove fitst and last element to avoid quote warningsk
-      (pop check-list)
-      (setq check-list (butlast check-list)))
+(defun flymake-languagetool--check-all (errors source-buffer)
+  "Check grammar ERRORS for SOURCE-BUFFER document."
+  (let (check-list)
+    (dolist (error errors)
+      (let-alist error
+        (push (flymake-make-diagnostic
+               source-buffer
+               (+ .offset 1)
+               (+ .offset .length 1)
+               :warning
+               (concat .message " [LanguageTool]"))
+              check-list)))
     check-list))
 
-(defun flymake-languagetool--cache-parse-result (output)
-  "Refresh cache buffer from OUTPUT."
-  (setq flymake-languagetool--output (car (flymake-languagetool--parse-json output))
-        flymake-languagetool--done-checking t)
-  (flymake-mode 1))
-
-(defun flymake-languagetool--send-process ()
-  "Send process to LanguageTool commandline-jar."
-  (if (not (file-exists-p flymake-languagetool-commandline-jar))
-      (user-error "Invalid commandline path: %s" flymake-languagetool-commandline-jar)
-    (when flymake-languagetool--done-checking
-      (setq flymake-languagetool--done-checking nil)  ; start flag
-      (flymake-languagetool--with-source-buffer
-       (let ((source (current-buffer)))
-         (flymake-languagetool--async-shell-command-to-string
-          (lambda (output)
-            (when (buffer-live-p source)
-              (with-current-buffer source (flymake-languagetool--cache-parse-result output))))
-          (format "echo %s | java -jar %s %s --json -b %s"
-                  (shell-quote-argument (s-replace "\n" " " (buffer-string)))
-                  flymake-languagetool-commandline-jar
-                  (if (stringp flymake-languagetool-language)
-                      (concat "-l " flymake-languagetool-language)
-                    "-adl")
-                  (if (stringp flymake-languagetool-args) flymake-languagetool-args ""))))))))
-
-(defun flymake-languagetool--start-timer ()
-  "Start the timer for grammar check."
-  (setq flymake-languagetool--source-buffer (current-buffer))
-  (when (timerp flymake-languagetool--timer) (cancel-timer flymake-languagetool--timer))
-  (setq flymake-languagetool--timer
-        (run-with-idle-timer flymake-languagetool-check-time nil
-                             #'flymake-languagetool--send-process)))
+(defun flymake-languagetool--output-to-errors (output source-buffer)
+  "Parse the JSON data from OUTPUT of LanguageTool. "
+  (let* ((objects nil)
```

Comment:
I think the variable is no longer needed?

Reply:
You are right. I've removed this.